### PR TITLE
planning: initialized hailing scenario

### PR DESCRIPTION
**Pull request by sjiang2018**

### CODE REVIEW

**Comment by YajiaZhang** [comment]
Maybe think about a better name for the scenario in the future.

File: modules/planning/scenarios/summon/summon_scenario_test.cc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+/**
+ * @file
+ **/
+
+#define protected public
```
Comment (Capri2014) [inline modules/planning/scenarios/summon/summon_scenario_test.cc]: Try avoid this practice

**Comment by Capri2014** [comment]
@sjiang2018 @YajiaZhang @yifeijiang @jmtao Do we plan to make it a scenario or an/a few stage for in-lane scenario?

File: modules/planning/scenarios/summon/summon_scenario_test.cc (hunk 1 above)
Comment (sjiang2018) [reply modules/planning/scenarios/summon/summon_scenario_test.cc]: Fixed. Thank you.

**Comment by YajiaZhang** [comment]
> @sjiang2018 @YajiaZhang @yifeijiang @jmtao Do we plan to make it a scenario or an/a few stage for in-lane scenario?

Let's make it a scenario first. From the functionality aspect, this should be the similar to pull-over scenario.

**Pull request merged**